src/stores/provider.js

```diff
@@ -0,0 +1,112 @@
+import { capital } from 'case'
+import { defineStore } from 'pinia'
+import { reactive } from '@nuxtjs/composition-api'
+
+import MediaProviderService from '~/data/media-provider-service'
+import { ALL_MEDIA, AUDIO, IMAGE, supportedMediaTypes } from '~/constants/media'
+import { warn } from '~/utils/console'
+import { useSearchStore } from '~/stores/search'
+
+const providerServices = {
+  [AUDIO]: MediaProviderService(AUDIO),
+  [IMAGE]: MediaProviderService(IMAGE),
+}
+
+/**
+ * Sorts providers by their source_name property.
+ * @param {import('../store/types').MediaProvider[]} data
+ * @returns {import('../store/types').MediaProvider[]}
+ */
+const sortProviders = (data) => {
+  return data.sort((sourceObjectA, sourceObjectB) => {
+    const nameA = sourceObjectA.source_name.toUpperCase()
+    const nameB = sourceObjectB.source_name.toUpperCase()
+    return nameA.localeCompare(nameB)
+  })
+}
```

Comment:
`sort` is not pure in JavaScript :disappointed: Should we spread `data` to avoid mutating it?

Adding `readonly` to array parameters would prevent these kinds of things I think.

```suggestion
/**
 * Sorts providers by their source_name property.
 * @param {readonly import('../store/types').MediaProvider[]} data
 * @returns {import('../store/types').MediaProvider[]}
 */
const sortProviders = (data) => {
  return [...data].sort((sourceObjectA, sourceObjectB) => {
    const nameA = sourceObjectA.source_name.toUpperCase()
    const nameB = sourceObjectB.source_name.toUpperCase()
    return nameA.localeCompare(nameB)
  })
}
```